modules/integration/src/test/java/io/ballerina/messaging/broker/integration/cluster/KillTest.java

```diff
@@ -0,0 +1,163 @@
+        properties.put(Context.INITIAL_CONTEXT_FACTORY, QPID_ICF);
+        properties.put(CF_NAME_PREFIX + CF_NAME, getTCPConnectionURL(username, password, hostnameOne, portOne,
+                hostnameTwo, portTwo));
+        properties.put(QUEUE_NAME_PREFIX + queueName, queueName);
+        ctx = new InitialContext(properties);
+
+        connFactory = (ConnectionFactory) ctx.lookup(CF_NAME);
+        Connection queueConnectionOne = connFactory.createConnection();
+        queueConnectionOne.start();
+
+        Session producerSession = queueConnectionOne.createSession(false, Session.AUTO_ACKNOWLEDGE);
+        Queue queue = producerSession.createQueue(queueName);
+        MessageProducer producer = producerSession.createProducer(queue);
+
+        int numberOfMessages = 10;
+        for (int i = 0; i < numberOfMessages; i++) {
+            producer.send(producerSession.createTextMessage("Test message " + i));
+        }
+        producerSession.close();
+        queueConnectionOne.close();
+    }
+
+    @Parameters({"broker-1-port", "broker-1-hostname"})
+    @Test
+    public void testNodeOneAvailableBeforeKill(String portOne, String hostnameOne) {
+        Assert.assertFalse(ClusterUtils.isPortAvailable(hostnameOne, Integer.parseInt(portOne)),
+                "Broker node one is not active");
+    }
+
+    @Parameters({"broker-2-port", "broker-2-hostname"})
+    @Test(dependsOnMethods = "testNodeOneAvailableBeforeKill")
+    public void testNodeTwoAvailableBeforeKill(String portTwo, String hostnameTwo) {
+        Assert.assertTrue(ClusterUtils.isPortAvailable(hostnameTwo, Integer.parseInt(portTwo)),
+                "Broker node two is not passive");
+    }
+
+    @Parameters({"broker-1-port", "broker-1-hostname"})
+    @Test(dependsOnMethods = "testNodeTwoAvailableBeforeKill")
+    public void testKillNodeOne(String portOne, String hostnameOne) throws IOException, InterruptedException {
+        ClusterUtils.killBrokerNode(portOne);
+        Awaitility.await().atMost(3000, TimeUnit.MILLISECONDS)
+                .pollInterval(5, TimeUnit.MILLISECONDS)
+                .until(() -> ClusterUtils.isPortAvailable(hostnameOne, Integer.parseInt(portOne)));
+        Assert.assertTrue(ClusterUtils.isPortAvailable(hostnameOne, Integer.parseInt(portOne)),
+                "Broker node one is not killed");
+    }
+
+    @Parameters({"broker-2-port", "broker-2-hostname"})
+    @Test(dependsOnMethods = "testKillNodeOne")
+    public void testNodeTwoActive(String portTwo, String hostnameTwo) throws NamingException, JMSException {
+        Awaitility.await().atMost(30, TimeUnit.SECONDS)
+                .pollInterval(3, TimeUnit.SECONDS)
+                .until(() -> !ClusterUtils.isPortAvailable(hostnameTwo, Integer.parseInt(portTwo)));
+        Assert.assertFalse(ClusterUtils.isPortAvailable(hostnameTwo, Integer.parseInt(portTwo)),
+                "Broker node two is not active");
+    }
+
+    @Test(dependsOnMethods = "testNodeTwoActive")
```

Comment:
Shall we add test method description here?

Reply:
Fixed. Added method descriptions